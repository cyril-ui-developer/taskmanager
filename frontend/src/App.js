import React, { useState} from "react";
import { Routes, Route } from "react-router-dom";
import "./App.css";
import TaskListPage from "./pages/TaskListPage";
import AddTaskPage from "./pages/AddTaskPage";
import Nav from "./components/Nav";

import {
  useGetTasksQuery,
  useAddTaskMutation,
  useUpdateTaskCompletedMutation,
  useDeleteTaskMutation,
} from "./redux/apiTaskSlice";

function App() {
  const [tasks, setTasks] = useState([]);
  const {
    data: tasksData,
    // isLoading,
    // isSuccess,
    //isError,
    // error,
  } =  useGetTasksQuery();

  const [addTask] =  useAddTaskMutation();
  const [updateTask] =   useUpdateTaskCompletedMutation();
  const [deleteTask] = useDeleteTaskMutation();

  const onAddTaskHandler = (formValues) => {
    const newTask = {
      ...formValues,
      id: tasks.length + 1,
      completed: false,
      active: false,
    };
    setTasks([...tasks, newTask]);
    addTask({ title: formValues.title, description: formValues.description });
  };

  const delTask = (taskId) => {
    setTasks((prev) => prev.filter((t) => t.id !== taskId));
    deleteTask(taskId);
  };

  const toggleTaskStatus = (taskId) => {
    const updatedTasks = tasksData.map((task) => {
      if (taskId === task.id) {
        updateTask({ id: taskId, completed: !task.completed });
        return { ...task, completed: !task.completed };
      }
      return task;
    });
    setTasks(updatedTasks);
  };

  return (
    <main className="flex items-center justify-center w-screen h-screen font-medium">
      <div className="flex flex-grow items-center justify-center h-full text-gray-600 bg-gray-100">
        <div className="max-w-full p-4 bg-white rounded-lg shadow-lg w-96">
          <section>
            <header className="flex items-center justify-between mb-6">
              <div className="flex items-center mr-4 mb-2">
                <input
                  type="checkbox"
                  checked={true}
                  readOnly
                  id="A3-yes"
                  name="A3-confirmation"
                  value="yes"
                  className="opacity-0 absolute h-8 w-8"
                />
                <div className="bg-white border-2 rounded-md border-blue-400 w-8 h-8 flex flex-shrink-0 justify-center items-center mr-2 focus-within:border-blue-500">
                  <svg
                    className="fill-current hidden w-3 h-3 text-blue-600 pointer-events-none"
                    version="1.1"
                    viewBox="0 0 17 12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g fill="none" fillRule="evenodd">
                      <g
                        transform="translate(-9 -11)"
                        fill="#1F73F1"
                        fillRule="nonzero"
                      >
                        <path d="m25.576 11.414c0.56558 0.55188 0.56558 1.4439 0 1.9961l-9.404 9.176c-0.28213 0.27529-0.65247 0.41385-1.0228 0.41385-0.37034 0-0.74068-0.13855-1.0228-0.41385l-4.7019-4.588c-0.56584-0.55188-0.56584-1.4442 0-1.9961 0.56558-0.55214 1.4798-0.55214 2.0456 0l3.679 3.5899 8.3812-8.1779c0.56558-0.55214 1.4798-0.55214 2.0456 0z" />
                      </g>
                    </g>
                  </svg>
                </div>
                <h4 className="font-semibold ml-1 text-lg">TaskManager</h4>
              </div>
              <Nav />
            </header>

            <Routes>
              <Route
                path="/"
                element={
                  <TaskListPage
                    tasks={tasksData}
                    onDeleteTask={delTask}
                    onToggleTaskStatus={toggleTaskStatus}
                  />
                }
              />
              <Route
                path="/add"
                element={<AddTaskPage onAddTaskHandler={onAddTaskHandler} />}
              />
            </Routes>
          </section>
        </div>
      </div>
    </main>
  );
}

export default App;
