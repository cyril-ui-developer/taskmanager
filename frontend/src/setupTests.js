// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';


import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { configureStore } from '@reduxjs/toolkit'
import { MemoryRouter } from 'react-router-dom'
import tasksSlice from './redux/taskSlice'

// [
//   { id: 1098754345678, title: "Lorem ipsum dolor sit", description: "Incididunt ut labore et dolore magna aliqua. Ut enim", completed: true,  active: false }, 
//   { id: 2009876543456, title: "Utenim ad minim ven", description: "Kexercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.", completed: false, active: false}
// ]
export function renderWithProviders(
  ui,
  {
    
    preloadedState ={},
    store = configureStore(
        tasksSlice,
      {preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>      <MemoryRouter>{children}</MemoryRouter></Provider>
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}