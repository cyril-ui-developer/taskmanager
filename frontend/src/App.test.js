import { screen} from '@testing-library/react';

import App from './App';
import { renderWithProviders } from './setupTests'

describe("renders", () => {
  it("will return a snapshot", () => {
    const view = renderWithProviders(
        <App />
    );
    expect(view).toMatchSnapshot();
  });

 it('renders app name', () => {
    renderWithProviders(<App />);
    const name = screen.getByText(/Taskie/i);
    expect(name).toBeInTheDocument();
  });
});

