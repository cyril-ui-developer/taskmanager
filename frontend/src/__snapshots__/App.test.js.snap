// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders will return a snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
            
      <main
        class="flex items-center justify-center w-screen h-screen font-medium"
      >
        <div
          class="flex flex-grow items-center justify-center h-full text-gray-600 bg-gray-100"
        >
          <div
            class="max-w-full p-4 bg-white rounded-lg shadow-lg w-96"
          >
            <section>
              <header
                class="flex items-center justify-between mb-6"
              >
                <div
                  class="flex items-center mr-4 mb-2"
                >
                  <input
                    checked=""
                    class="opacity-0 absolute h-8 w-8"
                    id="A3-yes"
                    name="A3-confirmation"
                    readonly=""
                    type="checkbox"
                    value="yes"
                  />
                  <div
                    class="bg-white border-2 rounded-md border-blue-400 w-8 h-8 flex flex-shrink-0 justify-center items-center mr-2 focus-within:border-blue-500"
                  >
                    <svg
                      class="fill-current hidden w-3 h-3 text-blue-600 pointer-events-none"
                      version="1.1"
                      viewBox="0 0 17 12"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="none"
                        fill-rule="evenodd"
                      >
                        <g
                          fill="#1F73F1"
                          fill-rule="nonzero"
                          transform="translate(-9 -11)"
                        >
                          <path
                            d="m25.576 11.414c0.56558 0.55188 0.56558 1.4439 0 1.9961l-9.404 9.176c-0.28213 0.27529-0.65247 0.41385-1.0228 0.41385-0.37034 0-0.74068-0.13855-1.0228-0.41385l-4.7019-4.588c-0.56584-0.55188-0.56584-1.4442 0-1.9961 0.56558-0.55214 1.4798-0.55214 2.0456 0l3.679 3.5899 8.3812-8.1779c0.56558-0.55214 1.4798-0.55214 2.0456 0z"
                          />
                        </g>
                      </g>
                    </svg>
                  </div>
                  <h4
                    class="font-semibold ml-1 text-lg"
                  >
                    Taskie
                  </h4>
                </div>
                <div
                  class="center"
                >
                  <a
                    class="hover:bg-blue-400 group flex items-center rounded-md bg-blue-500 text-white text-sm font-medium pl-2 pr-3 py-2 shadow-s"
                    href="/add"
                  >
                    <svg
                      aria-hidden="true"
                      class="mr-2"
                      fill="currentColor"
                      height="20"
                      width="20"
                    >
                      <path
                        d="M10 5a1 1 0 0 1 1 1v3h3a1 1 0 1 1 0 2h-3v3a1 1 0 1 1-2 0v-3H6a1 1 0 1 1 0-2h3V6a1 1 0 0 1 1-1Z"
                      />
                    </svg>
                    Add Task
                  </a>
                </div>
              </header>
              <h3>
                Total count: 
                0 tasks remaining
              </h3>
              <article
                class="flex items-start space-x-6 p-6"
              >
                <ul />
              </article>
            </section>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
          
    <main
      class="flex items-center justify-center w-screen h-screen font-medium"
    >
      <div
        class="flex flex-grow items-center justify-center h-full text-gray-600 bg-gray-100"
      >
        <div
          class="max-w-full p-4 bg-white rounded-lg shadow-lg w-96"
        >
          <section>
            <header
              class="flex items-center justify-between mb-6"
            >
              <div
                class="flex items-center mr-4 mb-2"
              >
                <input
                  checked=""
                  class="opacity-0 absolute h-8 w-8"
                  id="A3-yes"
                  name="A3-confirmation"
                  readonly=""
                  type="checkbox"
                  value="yes"
                />
                <div
                  class="bg-white border-2 rounded-md border-blue-400 w-8 h-8 flex flex-shrink-0 justify-center items-center mr-2 focus-within:border-blue-500"
                >
                  <svg
                    class="fill-current hidden w-3 h-3 text-blue-600 pointer-events-none"
                    version="1.1"
                    viewBox="0 0 17 12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <g
                        fill="#1F73F1"
                        fill-rule="nonzero"
                        transform="translate(-9 -11)"
                      >
                        <path
                          d="m25.576 11.414c0.56558 0.55188 0.56558 1.4439 0 1.9961l-9.404 9.176c-0.28213 0.27529-0.65247 0.41385-1.0228 0.41385-0.37034 0-0.74068-0.13855-1.0228-0.41385l-4.7019-4.588c-0.56584-0.55188-0.56584-1.4442 0-1.9961 0.56558-0.55214 1.4798-0.55214 2.0456 0l3.679 3.5899 8.3812-8.1779c0.56558-0.55214 1.4798-0.55214 2.0456 0z"
                        />
                      </g>
                    </g>
                  </svg>
                </div>
                <h4
                  class="font-semibold ml-1 text-lg"
                >
                  Taskie
                </h4>
              </div>
              <div
                class="center"
              >
                <a
                  class="hover:bg-blue-400 group flex items-center rounded-md bg-blue-500 text-white text-sm font-medium pl-2 pr-3 py-2 shadow-s"
                  href="/add"
                >
                  <svg
                    aria-hidden="true"
                    class="mr-2"
                    fill="currentColor"
                    height="20"
                    width="20"
                  >
                    <path
                      d="M10 5a1 1 0 0 1 1 1v3h3a1 1 0 1 1 0 2h-3v3a1 1 0 1 1-2 0v-3H6a1 1 0 1 1 0-2h3V6a1 1 0 0 1 1-1Z"
                    />
                  </svg>
                  Add Task
                </a>
              </div>
            </header>
            <h3>
              Total count: 
              0 tasks remaining
            </h3>
            <article
              class="flex items-start space-x-6 p-6"
            >
              <ul />
            </article>
          </section>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
